### PR TITLE
Clean up tda client option chain parsing

### PR DESCRIPTION
**Pull request by phall1**
2 commits here

1. Clean up the tda option chain parsing logic
2. Add black, isort and flake8 w/accompanying pre-commit hooks

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (ajhpark) [inline .vscode/settings.json:1]: Let's not commit IDE specific configurations. Could you add .vscode to .gitignore

**Review by ajhpark — CHANGES_REQUESTED** [review]
Thanks for the PR. It looks good mostly but please see my comment below. Changes for `tda_client.py` looks okay to me but maybe @wubr2000 would want to have a look as well?

File: .vscode/settings.json (hunk 1 above)
Comment (phall1) [reply .vscode/settings.json]: added in the latest revision

**Comment by ajhpark** [comment]
Thanks for your contribution. And sorry about the confusion with the force push to master. A previous contributor asked me to remove him from the commit history by rewriting it (bad practice, I know). Anyway, everything looks good, merging.

**Pull request merged**